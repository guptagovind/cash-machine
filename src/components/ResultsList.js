import React from 'react';
import uuid from "uuid";

import ResultListElement from './ResultListElement';

const ResultsList = ({results}) => (
  <div className="list">
    {
      (!Array.isArray(results))
        ?
        (
          <span>
                 Result: {results}
          </span>
        )
        :
        (
          <div>Result: [
            {
              results.map((result, index) => (
                <span key={uuid()}>
                      <ResultListElement result={result}/>
                  {results.length - 1 !== index && <span>, </span>}
                </span>
              ))
            }
            ]
          </div>
        )
    }
  </div>
);

export default ResultsList;
