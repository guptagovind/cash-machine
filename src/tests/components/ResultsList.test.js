import React from 'react';
import {shallow} from 'enzyme';
import ResultsList from '../../components/ResultsList';


let wrapper;
beforeEach(() => {
  wrapper = shallow(<ResultsList/>);
});

test('should render the resultsList', () => {
  expect(wrapper).toMatchSnapshot();
});
