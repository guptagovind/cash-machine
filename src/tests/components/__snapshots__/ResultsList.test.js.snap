// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the resultsList 1`] = `
<div
  className="list"
>
  <span>
    Result: 
  </span>
</div>
`;
